packages/@lwc/synthetic-shadow/src/faux-shadow/shadow-root.ts

```diff
@@ -44,14 +44,22 @@ import { getOuterHTML } from '../3rdparty/polymer/outer-html';
 import { retarget } from '../3rdparty/polymer/retarget';
 import { pathComposer } from '../3rdparty/polymer/path-composer';
 import { getInternalChildNodes, setNodeKey, setNodeOwnerKey } from './node';
-import { innerHTMLSetter } from '../env/element';
+import { innerHTMLSetter, setAttribute } from '../env/element';
 import { getOwnerDocument } from '../shared/utils';
 
 const { getHiddenField, setHiddenField, createFieldName } = fields;
 const ShadowRootResolverKey = '$shadowResolver$';
 const InternalSlot = createFieldName('shadowRecord', 'synthetic-shadow');
 const { createDocumentFragment } = document;
 
+/*
+ * Using a simple custom attribute is faster based on perf results:
+ * Setting attribute: https://jsperf.com/attribute-vs-data-attribute/1
+ * matches(selector): https://jsperf.com/matches-attribute-vs-data-attribute/1
+ * querySelector(): https://jsperf.com/query-by-attribute-vs-data-attribute/1
+ **/
+export const HostMarkerAttribute = 'lwc-host';
```

Comment:
this should include a generated number or some sore... otherwise we risk people using that attribute for other things.

Reply:
I think we should use `lwc:host="numeric-generated-value-per-app"`

Reply:
attribute named `lwc:host` cannot be used in a selector. So continuing to use `lwc-host` instead.